rclcpp/include/rclcpp/service.hpp

```diff
@@ -172,6 +172,7 @@ class Service : public ServiceBase
         (std::cerr << ss.str()).flush();
         rcl_reset_error();
       }
+      delete service_handle_;
```

Comment:
Certainly as the code stands we need it.  If you look at the (first) constructor for the Service() class, it does:

```
service_handle_ = new rcl_service_t;
```

Note that service_handle_ is not a shared_ptr<> or unique_ptr<>, and thus the programmer needs to explicitly call `delete service_handle_` somewhere (which was never being done).  `rcl_service_fini()` deals with freeing all of the *fields* of the rcl_service_t, but not the outer object itself.  This was all shown clearly when running the `test_executor` test from system_tests under valgrind.